src/base/index.es

```diff
@@ -1,20 +1,20 @@
 import { PureComponent } from 'react';
-import PropTypes from 'prop-types';
+import MapContext from '../map/context';
 
 
 class MapComponent extends PureComponent {
   componentDidMount() {
-    this.create(this.props, this.context);
-    this.context.map.addElement(this);
+    this.create();
+    this.context.addElement(this);
```

Comment:
might want to provide an escape hatch method `getContext` so you can override that if needed